enhancements/microshift/microshift-telemetry.md

```diff
@@ -0,0 +1,284 @@
+  pods, namespaces, services, etc.
+* cluster:usage:containers:sum. Number of active containers.
+* instance:etcd_object_counts:sum. Number of objects etcd contains.
+
+Combining metrics with their labels:
+| Metric | Labels |
+|:---|:---|
+|cluster:capacity_cpu_cores:sum|_id, instance, ostree_commit, version|
+|cluster:capacity_memory_bytes:sum|_id, instance, ostree_commit, version|
+|cluster:cpu_usage_cores:sum|_id, instance, ostree_commit|
+|cluster:memory_usage_bytes:sum|_id, instance, ostree_commit|
+|instance:etcd_object_counts:sum|_id, instance, ostree_commit|
+|cluster:usage:resources:sum|_id, instance, ostree_commit, resource|
+|cluster:usage:containers:sum|_id, instance, ostree_commit|
+
+Each metric, including a single sample and all labels, is under 250B in size.
+
+In order to keep metrics synchronized with those of OpenShift, in terms of API
+support, MicroShift will be using CI:
+* A job shall deploy MicroShift with Telemetry enabled.
+* A job will use the staging endpoint for Telemetry API. No added traffic
+  and no confusing data when querying.
+* At the end of the job a new check shall be performed: check there is data
+  belonging to that specific cluster in the staging environment.
+
+### Sending metrics
+OpenShift is sending metrics through this API every 4.5 minutes. For MicroShift
+deployments this might be a bit excessive due to resource usage, network
+traffic and usage patterns.
+
+Due to the Telemetry API characteristics, MicroShift will send data every hour.
+In terms of resource usage, each metric is about 250B in size and MicroShift
+will send 7 of them. Rounding up because of the multiple time series in the
+metrics measuring pods, resources and usage, each MicroShift telemetry message
+will be about ~2KB.
+
+When MicroShift starts it should send metrics once and then schedule them every
+hour.
+It will also send metrics when issued to stop, as part of the finish process.
+This helps in sending a last report and also hint when the cluster has been
+stopped.
+In order to introduce some entropy to alleviate traffic spikes in case of
+massive deployments (devices in a fleet, for example), the sending interval may
+be added a random offset to throw it off sync. Since the sending interval is
+1h, a 10% offset seems reasonable to spread out sending metrics over 12min.
+This is not configurable.
+
+As described above MicroShift will be using the [direct request](https://github.com/openshift/telemeter/tree/main?tab=readme-ov-file#metricsv1receive-endpoint-receive-metrics-in-prompbwriterequest-format-from-any-client)
+endpoint.
+Each metric must follow [Prometheus WriteRequest](https://github.com/prometheus/prometheus/blob/release-2.38/prompb/remote.proto#L22)
+format.
+
+Some metrics, like those that relate to hardware capabilities (which also
+include the MicroShift version) do not change unless there is a restart. These
+metrics can be sent only when MicroShift starts and/or stops.
```

Comment:
again this doesn't match the Telemetry expectations of receiving metrics at regular and (relatively) small intervals.

Reply:
Removed the possibly premature optimization. this is only about 500B in each write request.